File: challenges/1/solution.ts
function twoSum(nums: number[], target: number): number[] {
  let visited = new Map<number, number>();
  for (let i = 0; i < nums.length; i++) {
    const a = nums[i];
    const remainder = target - a;
    const dual = visited.get(remainder);
    visited.set(a, i);
    if (typeof dual === "number") return [dual, i];
  }
  return [];
}

File: challenges/1/solution.py
def twoSum(self, nums, target):
    
    if(len(nums)!=2):        
        pos = 0
        nums_dict = {}
        for num in nums:
            if target-num in nums_dict:
                # bingo!
                return [pos, nums_dict[target-num]]
            nums_dict[num]=pos    
            pos+=1

    # best case
    return [0,1]